src/Hangfire.PostgreSql/Utils/AutoResetEventRegistry.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Hangfire.PostgreSql.Utils
+{
+  /// <summary>
+  /// Represents a registry for managing AutoResetEvent instances using event keys.
+  /// </summary>
+  public class AutoResetEventRegistry
+  {
+    private readonly ConcurrentDictionary<string, AutoResetEvent> _events = new();
+
+    /// <summary>
+    /// Retrieves the wait handles associated with the specified event keys.
+    /// </summary>
+    /// <param name="eventKeys">The event keys.</param>
+    /// <returns>An enumerable of wait handles.</returns>
+    public IEnumerable<WaitHandle> GetWaitHandles(IEnumerable<string> eventKeys)
+    {
+      foreach (string eventKey in eventKeys)
+      {
+        if (_events.TryGetValue(eventKey, out AutoResetEvent handle))
+        {
+          yield return handle;
+        }
+        else
+        {
+          AutoResetEvent newHandle = _events.GetOrAdd(eventKey, new AutoResetEvent(false));
+          yield return newHandle;
+        }
```

Comment:
Why need the `if` statement here? It should work the same if you keep only `else` branch.

```suggestion
      {
          AutoResetEvent handle = _events.GetOrAdd(eventKey, new AutoResetEvent(false));
          yield return handle;
       }
```

Reply:
You should be using overload with `Func` parameter.

```suggestion
      {
          AutoResetEvent newHandle = _events.GetOrAdd(eventKey, _ => new AutoResetEvent(false));
          yield return newHandle;
```